tests/sample/tail_call_max_exceed.c

```diff
@@ -0,0 +1,147 @@
+DECLARE_BIND_TAIL_FUNC(3)
+DECLARE_BIND_TAIL_FUNC(4)
+DECLARE_BIND_TAIL_FUNC(5)
+DECLARE_BIND_TAIL_FUNC(6)
+DECLARE_BIND_TAIL_FUNC(7)
+DECLARE_BIND_TAIL_FUNC(8)
+DECLARE_BIND_TAIL_FUNC(9)
+DECLARE_BIND_TAIL_FUNC(10)
+DECLARE_BIND_TAIL_FUNC(11)
+DECLARE_BIND_TAIL_FUNC(12)
+DECLARE_BIND_TAIL_FUNC(13)
+DECLARE_BIND_TAIL_FUNC(14)
+DECLARE_BIND_TAIL_FUNC(15)
+DECLARE_BIND_TAIL_FUNC(16)
+DECLARE_BIND_TAIL_FUNC(17)
+DECLARE_BIND_TAIL_FUNC(18)
+DECLARE_BIND_TAIL_FUNC(19)
+DECLARE_BIND_TAIL_FUNC(20)
+DECLARE_BIND_TAIL_FUNC(21)
+DECLARE_BIND_TAIL_FUNC(22)
+DECLARE_BIND_TAIL_FUNC(23)
+DECLARE_BIND_TAIL_FUNC(24)
+DECLARE_BIND_TAIL_FUNC(25)
+DECLARE_BIND_TAIL_FUNC(26)
+DECLARE_BIND_TAIL_FUNC(27)
+DECLARE_BIND_TAIL_FUNC(28)
+DECLARE_BIND_TAIL_FUNC(29)
+DECLARE_BIND_TAIL_FUNC(30)
+DECLARE_BIND_TAIL_FUNC(31)
+DECLARE_BIND_TAIL_FUNC(32)
+DECLARE_BIND_TAIL_FUNC(33)
+DECLARE_BIND_TAIL_FUNC(34)
+
+struct
+{
+    __uint(type, BPF_MAP_TYPE_PROG_ARRAY);
+    __type(key, uint32_t);
+    __uint(max_entries, MAX_TAIL_CALL_CNT + 2);
+    __array(values, bind_action_t(bind_md_t* ctx));
+} bind_tail_call_map SEC(".maps") = {
+    .values = {
+        bind_test_callee0,  bind_test_callee1,  bind_test_callee2,  bind_test_callee3,  bind_test_callee4,
+        bind_test_callee5,  bind_test_callee6,  bind_test_callee7,  bind_test_callee8,  bind_test_callee9,
+        bind_test_callee10, bind_test_callee11, bind_test_callee12, bind_test_callee13, bind_test_callee14,
+        bind_test_callee15, bind_test_callee16, bind_test_callee17, bind_test_callee18, bind_test_callee19,
+        bind_test_callee20, bind_test_callee21, bind_test_callee22, bind_test_callee23, bind_test_callee24,
+        bind_test_callee25, bind_test_callee26, bind_test_callee27, bind_test_callee28, bind_test_callee29,
+        bind_test_callee30, bind_test_callee31, bind_test_callee32, bind_test_callee33, bind_test_callee34,
+    }};
+
+DEFINE_BIND_TAIL_FUNC(0)
+DEFINE_BIND_TAIL_FUNC(1)
+DEFINE_BIND_TAIL_FUNC(2)
+DEFINE_BIND_TAIL_FUNC(3)
+DEFINE_BIND_TAIL_FUNC(4)
+DEFINE_BIND_TAIL_FUNC(5)
+DEFINE_BIND_TAIL_FUNC(6)
+DEFINE_BIND_TAIL_FUNC(7)
+DEFINE_BIND_TAIL_FUNC(8)
+DEFINE_BIND_TAIL_FUNC(9)
+DEFINE_BIND_TAIL_FUNC(10)
+DEFINE_BIND_TAIL_FUNC(11)
+DEFINE_BIND_TAIL_FUNC(12)
+DEFINE_BIND_TAIL_FUNC(13)
+DEFINE_BIND_TAIL_FUNC(14)
+DEFINE_BIND_TAIL_FUNC(15)
+DEFINE_BIND_TAIL_FUNC(16)
+DEFINE_BIND_TAIL_FUNC(17)
+DEFINE_BIND_TAIL_FUNC(18)
+DEFINE_BIND_TAIL_FUNC(19)
+DEFINE_BIND_TAIL_FUNC(20)
+DEFINE_BIND_TAIL_FUNC(21)
+DEFINE_BIND_TAIL_FUNC(22)
+DEFINE_BIND_TAIL_FUNC(23)
+DEFINE_BIND_TAIL_FUNC(24)
+DEFINE_BIND_TAIL_FUNC(25)
+DEFINE_BIND_TAIL_FUNC(26)
+DEFINE_BIND_TAIL_FUNC(27)
+DEFINE_BIND_TAIL_FUNC(28)
+DEFINE_BIND_TAIL_FUNC(29)
+DEFINE_BIND_TAIL_FUNC(30)
+DEFINE_BIND_TAIL_FUNC(31)
+DEFINE_BIND_TAIL_FUNC(32)
+DEFINE_BIND_TAIL_FUNC(33)
+
+bind_hook_t bind_test_caller;
+
+SEC("bind")
+bind_action_t
+bind_test_caller(bind_md_t* ctx)
+{
+    bpf_printk("bind_test_caller: Start Tail call index %d\n", 0);
```

Comment:
```suggestion
    bpf_printk("bind_test_caller: Start tail call index %d\n", 0);
```

Reply:
Also what's the point of passing a constant to %d instead of just putting 0 in the string?